frontend/src/components/charts/DurationSummaryChart.tsx

```diff
@@ -0,0 +1,128 @@
+import { useTheme } from "@mui/material/styles/";
+import ReactEcharts from "echarts-for-react";
+
+import { Difficulty } from "~shared/types/base/index";
+
+type DurationSummaryProps = {
+  durationSummary:
+    | {
+        difficulty: Difficulty;
+        timetaken: number;
+        date: Date;
+      }[]
+    | undefined;
+};
+
+export const DurationSummaryChart = ({
+  durationSummary,
+}: DurationSummaryProps) => {
+  const theme = useTheme();
+
+  const processDates = (
```

Comment:
Are we removing dates that do not have any records? Not sure if it would make it misleading if the difference in x-axis values isn't consistent?

Reply:
Lets use dates across 30 days